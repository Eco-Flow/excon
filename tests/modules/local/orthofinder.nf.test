nextflow_process {

    name "Test Process ORTHOFINDER"
    script "modules/local/orthofinder.nf"
    process "ORTHOFINDER"

    test("Should run without failures") {

        when {
            params {
                // define parameters here:
                clean = false
                outdir = "results"
            }
            process {
                """
                // define inputs of the process here. Example:
                Channel.fromPath( "${projectDir}/data/orthofinder_input/*.fa" ).collect().set{ in_channel } 
                input[0] = in_channel
                """
            }
        }

        then {
            //check if test case succeeded
            assert process.success
            //analyze trace file
            //assert process.trace.tasks().size() == 1 
            with(process.out) {
              //checks one file exists in orthologues channel
              assert orthologues.size() == 1
              //checks one file exists in species tree channel
              assert speciestree.size() == 1
              //checks output files aren't corrupted
              assert path(orthologues[0]).md5 == 'd05504abd3f1f7437e606c8a0a41d6c4'
              assert path(versions[0]).md5 == '76d821b267ab1de6f1fcc8840eb7dca9'

            }


            //Cannot assert snapsnot, as output differs each time we run it due to random seeding. Means each phylogentic tree scores are slightly different.
        }
    }
}
