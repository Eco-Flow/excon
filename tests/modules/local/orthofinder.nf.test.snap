{
    "orthofinder": {
        "content": [
            [
                [
                    [
                        [
                            "Orthogroups.tsv:md5,7d1c15f8d74eda5e11fabc8aa8ae0b0a"
                        ],
                        [
                            "N0.tsv:md5,e040bdfc8539ce3407dbfff6459f3736"
                        ],
                        [
                            "SpeciesTree_rooted_node_labels.txt:md5,3adb133e3e47b8c6dc17d1b563355790"
                        ]
                    ]
                ],
                "versions.yml:md5,76d821b267ab1de6f1fcc8840eb7dca9"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T19:04:17.165482591"
    }
}